### PR TITLE
Extend first-party policy hint resolution (9.7.1)

### PR DESCRIPTION
**Pull request by leynos**
## Summary

- Implements roadmap item 9.7.1 from `docs/execplans/9-7-1-first-party-policy-hint-resolution.md`.
- Adds `rstest` regression coverage proving first-party harness paths resolve to the same `TestAttributeHint` values as their matching attribute-policy paths.
- Extends the unknown-path matrix for exact matching, including wrong first-party-looking prefixes and extra path segments.
- Fixes trybuild support staging so harness compile fixtures can copy support files into destination parent chains that do not yet exist.
- Marks roadmap item 9.7.1 done after validation.

## Validation

- `cargo test -p rstest-bdd-policy`
- `cargo test -p rstest-bdd-macros codegen::scenario`
- `cargo test -p rstest-bdd-harness trybuild_staging`
- `cargo test -p rstest-bdd-harness-gpui gpui_macro_fixtures_compile --features native-gpui-tests`
- `make check-fmt`
- `make lint`
- `make test`
- `make markdownlint`
- `make nixie`

## Notes

`make test` now passes after the trybuild staging fix: nextest ran 1391 tests with 1391 passed and 7 skipped, and the Python publish-check suite passed 62 tests.

## Summary by Sourcery

Extend first-party policy hint resolution documentation and regression coverage, and fix trybuild staging to handle missing destination parent chains while preserving overlap safety.

Enhancements:
- Refine copy_dir_tree canonicalization to resolve destinations via the nearest existing ancestor and replay missing path components while correctly handling parent-directory segments.
- Expose and document shared first-party path constants in rstest-bdd-policy so macros, harness adapters, and docs use consistent canonical policy locations.

Documentation:
- Document shared first-party harness and attribute-policy path constants and resolver helpers, including exact-match semantics and usage guidance.
- Add an execution plan document for roadmap item 9.7.1, detailing constraints, risks, progress, decisions, and validation for first-party policy hint resolution.
- Clarify developer guidance on trybuild staging behaviour, including how copy_dir_tree rejects overlapping trees with missing destination parents.

Tests:
- Add integration tests for copy_dir_tree to cover creating missing destination parent chains and rejecting overlapping destinations that resolve back into the source tree.
- Extend rstest-based coverage ensuring first-party harness paths resolve to the same TestAttributeHint as their corresponding attribute-policy paths, and that non-exact or third-party-looking paths return None.

Chores:
- Mark roadmap item 9.7.1 as completed with delivery notes after validation and gating.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/rstest-bdd-harness/src/trybuild_staging/mod.rs](https://github.com/leynos/rstest-bdd/pull/488/files/e90b8cac925d5ee1258af533768869558a1c1996#diff-5d3ec818898dc92e7455ef2a75062ec2d46ca0b04fc569027bee136b851f4ffe)

Comment on lines +93 to +122

~~~rust
fn canonical_missing_destination(destination: &Path) -> io::Result<PathBuf> {
    let mut missing_components = Vec::new();
    let mut current = destination;

    loop {
        match fs::canonicalize(current) {
            Ok(mut path) => {
                for component in missing_components.iter().rev() {
                    path.push(component);
                }
                return Ok(path);
            }
            Err(err) if err.kind() == io::ErrorKind::NotFound => {
                if let Some(name) = current.file_name() {
                    missing_components.push(name.to_os_string());
                }
                if let Some(parent) = current.parent().filter(|path| !path.as_os_str().is_empty()) {
                    current = parent;
                } else {
                    let mut path = fs::canonicalize(std::env::current_dir()?)?;
                    for component in missing_components.iter().rev() {
                        path.push(component);
                    }
                    return Ok(path);
                }
            }
            Err(err) => return Err(err),
        }
    }
}
~~~

❌ New issue: [Bumpy Road Ahead](https://codescene.io/projects/69391/delta?repo-id=1163913&review-id=488&biomarker=Bumpy+Road+Ahead&filename=crates%2Frstest-bdd-harness%2Fsrc%2Ftrybuild_staging%2Fmod.rs&method=canonical_missing_destination)
canonical_missing_destination has 2 blocks with nested conditional logic. Any nesting of 2 or deeper is considered. Threshold is 2 blocks per function

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/rstest-bdd-harness/src/trybuild_staging/mod.rs](https://github.com/leynos/rstest-bdd/pull/488/files/d82485c51aa83e5d23dfad3d9c742fa0d364bb77#diff-5d3ec818898dc92e7455ef2a75062ec2d46ca0b04fc569027bee136b851f4ffe)

Comment on lines +100 to +121

~~~rust
fn canonical_missing_destination(destination: &Path) -> io::Result<PathBuf> {
    let mut missing_components: Vec<std::ffi::OsString> = Vec::new();

    for ancestor in destination.ancestors() {
        if ancestor.as_os_str().is_empty() {
            break;
        }
        match fs::canonicalize(ancestor) {
            Ok(base) => return Ok(append_missing_components(base, &missing_components)),
            Err(err) if err.kind() == io::ErrorKind::NotFound => {
                if let Some(name) = ancestor.file_name() {
                    missing_components.push(name.to_os_string());
                }
            }
            Err(err) => return Err(err),
        }
    }

    // No existing ancestor found; resolve against the current working directory.
    let base = fs::canonicalize(std::env::current_dir()?)?;
    Ok(append_missing_components(base, &missing_components))
}
~~~

❌ New issue: [Deep, Nested Complexity](https://codescene.io/projects/69391/delta?repo-id=1163913&review-id=488&biomarker=Deep%2C+Nested+Complexity&filename=crates%2Frstest-bdd-harness%2Fsrc%2Ftrybuild_staging%2Fmod.rs&method=canonical_missing_destination)
canonical_missing_destination has a nested complexity depth of 4, threshold = 4

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 errors, 2 warnings, 1 inconclusive)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | New test only verifies successful copy with missing destination parents, not overlap detection. Test duplicates setup instead of using make\_src\_dst\_scaffold() helper flagged in review comments. | Add test cases verifying overlap rejection still works with missing destination parents. Refactor new test to use make\_src\_dst\_scaffold() fixture to avoid setup code drift. |
| Unit Architecture | ❌ Error | The new test duplicates tempdir/src setup from existing `make_src_dst_scaffold()` helper instead of reusing it, creating maintenance drift risk and violating the pattern used by other tests. | Refactor the test to route through `make_src_dst_scaffold()` or a dedicated fixture to reuse shared test infrastructure, as the review comment requires. This centralises test harness setup logic. |
| Security And Privacy | ❌ Error | Hardcoded personal directory path in docs/execplans/9-7-1 file (`/home/leynos/...`). Roadmap note lacks explicit ADR exception statement. | Replace path with portable `$(git rev-parse --show-toplevel)`. Update roadmap 9.7.1 to state delivery occurred under maintainer authorisation while ADR-008 remained Proposed. |
| Developer Documentation | ⚠️ Warning | Execplan has hardcoded path; roadmap missing ADR-008 contingency note; new public APIs undocumented in developers-guide. | Fix hardcoded path in execplan; add ADR-008 authorization clause to roadmap; document new rstest-bdd-policy APIs in developers-guide; update developers-guide staging section on missing parents. |
| Testing (Unit And Behavioural) | ⚠️ Warning | Test duplicates setup code from `make_src_dst_scaffold()`. Other tests use this fixture successfully, but the new test manually reinvents setup, risking maintenance drift. | Use `make_src_dst_scaffold()` as the fixture base like `copy_dir_tree_replaces_existing_directory`. Construct custom `dst` path for the missing-chain scenario inside the test body. |
| User-Facing Documentation | ❓ Inconclusive | No result was produced after verification. Marking as INCONCLUSIVE. | Re-run the check or adjust instructions to produce a final result. |

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/rstest-bdd-harness/src/trybuild_staging/tests.rs](https://github.com/leynos/rstest-bdd/pull/488/files/c4b0ef4a49d0b283192c7cdd04d32ef77328e730#diff-02ff62dce2e9c0676949dfd4983db1582989ae98e2b97ef496ecc74132f3fb43)

Comment on file

~~~rust
    assert!(dst.join("f.txt").exists());
}

#[test]
~~~

❌ Getting worse: [Code Duplication](https://codescene.io/projects/69391/delta?repo-id=1163913&review-id=488&biomarker=Code+Duplication&filename=crates%2Frstest-bdd-harness%2Fsrc%2Ftrybuild_staging%2Ftests.rs&method=copy_dir_tree_rejects_destination_resolved_through_missing_parent_dir)
introduced similar code in: copy_dir_tree_rejects_destination_resolved_through_missing_parent_dir

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/rstest-bdd-harness/src/trybuild_staging/tests.rs](https://github.com/leynos/rstest-bdd/pull/488/files/0841413e04988c9138fc423cfbfb539b8b7e6b7d#diff-02ff62dce2e9c0676949dfd4983db1582989ae98e2b97ef496ecc74132f3fb43)

Comment on file

~~~rust
    assert!(dst.join("f.txt").exists());
}

#[test]
~~~

❌ Getting worse: [Code Duplication](https://codescene.io/projects/69391/delta?repo-id=1163913&review-id=488&biomarker=Code+Duplication&filename=crates%2Frstest-bdd-harness%2Fsrc%2Ftrybuild_staging%2Ftests.rs&method=)
introduced similar code in: copy_dir_tree_rejects_destination_inside_missing_parent_chain,copy_dir_tree_rejects_destination_resolved_through_missing_parent_dir

File: crates/rstest-bdd-harness/src/trybuild_staging/mod.rs [hunk 1]
```diff
@@ -84,20 +84,40 @@ fn canonical_destination_for_overlap(destination: &Path) -> io::Result<PathBuf>
...
+    let mut missing_components: Vec<std::ffi::OsString> = Vec::new();
+
+    for ancestor in destination
+        .ancestors()
+        .take_while(|p| !p.as_os_str().is_empty())
+    {
+        match fs::canonicalize(ancestor) {
+            Ok(base) => return Ok(append_missing_components(base, &missing_components)),
+            Err(err) if err.kind() == io::ErrorKind::NotFound => {
+                missing_components.extend(ancestor.file_name().map(std::ffi::OsStr::to_os_string));
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd-harness/src/trybuild_staging/mod.rs]: @coderabbitai has this now been resolved?

Use sourcegraph analysis to determine your answer.

If the comment has been resolved, please mark it as resolved using the API. Otherwise, please provide an AI agent prompt for the work remaining.

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response. Ensure that this is validated against the current version of the codegraph.

If further refinement to address this finding would be deleterious, please supply a clear explanatory one to two paragraph markdown message I can paste into the CodeScene web ui's diagnostic suppression function so this diagnostic can be silenced.

[crates/rstest-bdd-harness/src/trybuild_staging/tests.rs](https://github.com/leynos/rstest-bdd/pull/488/files/ce9ed063071a2b4aa9043f6018ed6ab7d5baf176#diff-02ff62dce2e9c0676949dfd4983db1582989ae98e2b97ef496ecc74132f3fb43)

Comment on file

~~~rust
    (root, src, dst)
}

struct OverlapCheckStaging {
~~~

❌ Getting worse: [Code Duplication](https://codescene.io/projects/69391/delta?repo-id=1163913&review-id=488&biomarker=Code+Duplication&filename=crates%2Frstest-bdd-harness%2Fsrc%2Ftrybuild_staging%2Ftests.rs&method=)
introduced similar code in: copy_dir_tree_rejects_destination_inside_missing_parent_chain,copy_dir_tree_rejects_destination_resolved_through_missing_parent_dir

**Comment by leynos** [comment]
@coderabbitai have the following now been resolved?

## ❌ Still unresolved (1 error, 1 inconclusive)

### Developer Documentation ⚠️

`docs/developers-guide.md` (456 lines) contains no documentation for:

- `trybuild_staging::copy_dir_tree` behaviour when destination parent chains are missing
- `STD_HARNESS_PATH`, `TOKIO_HARNESS_PATH`, `GPUI_HARNESS_PATH`, `GPUI_ATTRIBUTE_POLICY_PATH`, `DEFAULT_ATTRIBUTE_POLICY_PATH`
- `resolve_test_attribute_hint_for_policy_path` and `resolve_test_attribute_hint_for_harness_path`

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

~~~
Run leynos/shared-actions/.github/actions/generate-coverage@aebb3f5b831102e2a10ef909c83d7d50ea86c332
  with:
    output-path: lcov.info
    format: lcov
    features: rstest-bdd/diagnostics rstest-bdd-macros/compile-time-validation rstest-bdd-server/test-support
    with-default-features: true
    use-cargo-nextest: false
    with-ratchet: false
    baseline-rust-file: .coverage-baseline.rust
    baseline-python-file: .coverage-baseline.python
    with-cucumber-rs: false
  env:
    CARGO_TERM_COLOR: always
    CS_ACCESS_TOKEN: 
    CODESCENE_CLI_SHA256: 
    RUSTFLAGS: -D warnings
    RUSTDOCFLAGS: -D warnings
    CARGO_INCREMENTAL: 0
    CARGO_PROFILE_DEV_DEBUG: 0
    RUST_BACKTRACE: short
    CARGO_UNSTABLE_SPARSE_REGISTRY: true
    CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
    UV_TOOL_BIN_DIR: D:\a\_temp\uv-tool-bin-dir
    UV_TOOL_DIR: D:\a\_temp\uv-tool-dir
    UV_CACHE_DIR: D:\a\_temp\setup-uv-cache
    SCCACHE_PATH: C:\hostedtoolcache\windows\sccache\0.15.0\x64/sccache
    ACTIONS_CACHE_SERVICE_V2: on
    ACTIONS_RESULTS_URL: https://results-receiver.actions.githubusercontent.com/
    ACTIONS_RUNTIME_TOKEN: ***
    UV_PYTHON_INSTALL_DIR: D:\a\_temp\uv-python-dir
...
error: struct `SymlinkInSourceStaging` is never constructed
Error:    --> crates\rstest-bdd-harness\src\trybuild_staging\tests.rs:226:8
    |
226 | struct SymlinkInSourceStaging {
    |        ^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D dead-code` implied by `-D warnings`
    = help: to override `-D warnings` add `#[expect(dead_code)]` or `#[allow(dead_code)]`

error: process didn't exit successfully: `'C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin\rustc.exe' --crate-name rstest_bdd_harness --edition=2024 'crates\rstest-bdd-harness\src\lib.rs' --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --emit=dep-info,link -C opt-level=1 -C embed-bitcode=no '--warn=clippy::pedantic' '--deny=clippy::use_self' '--deny=clippy::unwrap_used' --forbid=unsafe_code '--deny=clippy::string_slice' '--deny=clippy::string_lit_as_bytes' '--deny=clippy::print_stdout' '--deny=clippy::print_stderr' '--deny=clippy::or_fun_call' '--deny=clippy::option_if_let_else' --deny=missing_docs --deny=missing_crate_level_docs '--deny=clippy::little_endian_bytes' '--deny=clippy::integer_division_remainder_used' '--deny=clippy::integer_division' '--deny=clippy::indexing_slicing' '--deny=clippy::host_endian_bytes' '--deny=clippy::float_arithmetic' '--deny=clippy::expect_used' '--deny=clippy::dbg_macro' '--deny=clippy::blanket_clippy_restriction_lints' '--deny=clippy::big_endian_bytes' '--deny=clippy::allow_attributes_without_reason' '--deny=clippy::allow_attributes' -C debug-assertions=on --test --check-cfg 'cfg(docsrs,test)' --check-cfg 'cfg(feature, values())' -C metadata=3d77d0a3d8255f9f -C extra-filename=-ade593d017a37ac0 --out-dir 'D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target\debug\deps' -C strip=debuginfo -L 'dependency=D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target\debug\deps' --extern 'cargo_metadata=D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target\debug\deps\libcargo_metadata-494e3ad7a68698c0.rlib' --extern 'insta=D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target\debug\deps\libinsta-6879e1cc30a293b6.rlib' --extern 'proptest=D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target\debug\deps\libproptest-d73156a10dc80f1e.rlib' --extern 'rstest=D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target\debug\deps\librstest-3a8ad28607e0a44e.rlib' --extern 'temp_env=D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target\debug\deps\libtemp_env-b5ab9738b1df1fe4.rlib' --extern 'tempfile=D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target\debug\deps\libtempfile-a8750bbd4100292a.rlib' --extern 'thiserror=D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target\debug\deps\libthiserror-25374ad208fe6c23.rlib' --extern 'tracing=D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target\debug\deps\libtracing-75b7266c9ce9e2a9.rlib' -D warnings -L 'native=C:\Users\runneradmin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\windows_x86_64_msvc-0.52.6\lib' -C instrument-coverage --cfg=coverage` (exit code: 1)
error: could not compile `rstest-bdd-harness` (lib test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: process didn't exit successfully: `'C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin\cargo.exe' test --tests --manifest-path 'D:\a\rstest-bdd\rstest-bdd\Cargo.toml' --target-dir 'D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target' --workspace --features 'rstest-bdd/diagnostics rstest-bdd-macros/compile-time-validation rstest-bdd-server/test-support'` (exit code: 101)
cargo llvm-cov --workspace --summary-only --features 'rstest-bdd/diagnostics rstest-bdd-macros/compile-time-validation rstest-bdd-server/test-support' --lcov --output-path lcov.info failed with code 101
Error: Process completed with exit code 101.
~~~

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (1 error, 1 warning, 1 inconclusive)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Unit Architecture | ❌ Error | Architecture sound: clear query/command separation, explicit fallibility, single responsibility. Issue: `make fmt` command in validation sequence must be removed per review comment. | Remove `make fmt` block lines 549–550 from execplan validation gates. Keep check-fmt, lint, test only. |
| Developer Documentation | ⚠️ Warning | Developer documentation comprehensive; roadmap item complete; ADR note added. Execplan validation still contains `make fmt`, contradicting review request to remove it. | Remove `make fmt` from execplan lines 546–557; keep `check-fmt`, `lint`, `test` only. |
| Testing (Overall) | ❓ Inconclusive | No result was produced after verification. Marking as INCONCLUSIVE. | Re-run the check or adjust instructions to produce a final result. |

File: docs/developers-guide.md [hunk 2]
```diff
@@ -417,6 +444,60 @@ Use the same pattern in hand-written tests instead of bare `.unwrap()`. This
...
+
+The following `&[&str]` constants identify the known first-party harness
+and attribute-policy types:
+
+| Constant | Path segments |
+| --- | --- |
+| `STD_HARNESS_PATH` | `["rstest_bdd_harness", "StdHarness"]` |
+| `TOKIO_HARNESS_PATH` | `["rstest_bdd_harness_tokio", "TokioHarness"]` |
+| `GPUI_HARNESS_PATH` | `["rstest_bdd_harness_gpui", "GpuiHarness"]` |
+| `DEFAULT_ATTRIBUTE_POLICY_PATH` | `["rstest_bdd_harness", "StdAttributePolicy"]` |
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/developers-guide.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

~~~
Please address the comments from this code review:

## Overall Comments
- The new `canonical_missing_destination`/`append_missing_components`/`missing_component_for` logic for reconstructing missing destination paths is fairly intricate and relies on manually re-appending `..` via `OsString`; consider simplifying this by working directly with `std::path::Component` (push/pop on a component stack) or at least tightening the implementation (e.g., avoid `extend(missing_component_for(...))` on `Option` and explicitly document/guard edge cases like all-`ParentDir` paths) to reduce the risk of subtle cross-platform path-normalization bugs.

## Individual Comments

### Comment 1
<location path="crates/rstest-bdd-harness/src/trybuild_staging/tests.rs" line_range="201-210" />
<code_context>
+fn copy_dir_tree_rejects_missing_tail_overlap_destinations(
</code_context>
<issue_to_address>
**suggestion (testing):** Also assert that no destination tree is created when an overlapping destination is rejected.

To strengthen this test against regressions, please also assert that the destination path is not created (or remains empty) when `copy_dir_tree` returns an error, e.g.:

```rust
assert!(!dst.exists(), "dst should not be created on overlap rejection");
```

(or, if the directory is always created, assert that `dst.read_dir()` is empty). This ensures we don’t silently copy any files before rejecting the overlap.

Suggested implementation:

```rust
    assert!(result.is_err());

    // Ensure that the destination path does not get created when overlap is rejected.
    // This guards against partially-copied trees before erroring.
    assert!(
        !dst.exists(),
        "dst should not be created on overlap rejection"
    );

```

I only see the beginning of `copy_dir_tree_rejects_missing_tail_overlap_destinations`, so I assumed there is a local `result` from calling `copy_dir_tree` and an existing `assert!(result.is_err());` in this test.

If that assertion currently does not exist, or if the variable name differs, you should:
1. Locate the place where `copy_dir_tree` is invoked and its `Result` is asserted as an error.
2. Immediately after that assertion, insert the new `assert!(!dst.exists(), ...)` so that it checks the filesystem state *after* the error is returned.
3. If your implementation always creates `dst` before the overlap check, then instead assert that `dst.read_dir()` is empty:
   ```rust
   let mut entries = dst.read_dir().expect("dst should be readable when created");
   assert!(
       entries.next().is_none(),
       "dst should remain empty on overlap rejection"
   );
   ```
   and adjust the SEARCH/REPLACE block accordingly.
</issue_to_address>

### Comment 2
<location path="crates/rstest-bdd-harness/src/trybuild_staging/mod.rs" line_range="115" />
<code_context>
+        })
+}
+
+fn canonical_missing_destination(destination: &Path) -> io::Result<PathBuf> {
+    let mut missing_components: Vec<std::ffi::OsString> = Vec::new();
+
</code_context>
<issue_to_address>
**issue (complexity):** Consider refactoring `canonical_missing_destination` to use an existing-ancestor-plus-tail approach and remove the custom `missing_components`/`OsString` reconstruction utilities.

You can simplify `canonical_missing_destination` by eliminating the reversed `missing_components` buffer and the custom `missing_component_for` logic, and instead work with a straightforward “existing ancestor + tail” approach.

Concretely:

- Walk ancestors until `canonicalize` succeeds (or fall back to `current_dir`).
- Record which ancestor succeeded.
- Compute the tail from that ancestor to `destination`.
- Rebuild the final path by applying the tail components (including `..`) directly, without a reversed `Vec<OsString>`.

This removes the need for both `append_missing_components` and `missing_component_for` while preserving behavior.

Example refactor:

```rust
fn apply_tail(mut base: PathBuf, tail: &Path) -> PathBuf {
    use std::path::Component;

    for component in tail.components() {
        match component {
            Component::ParentDir => {
                base.pop();
            }
            Component::Normal(name) => {
                base.push(name);
            }
            // Skip CurDir / RootDir / Prefix here; they shouldn't appear in a relative tail
            _ => {}
        }
    }
    base
}

fn canonical_missing_destination(destination: &Path) -> io::Result<PathBuf> {
    use std::io::ErrorKind;

    let mut found_base: Option<(PathBuf, &Path)> = None;

    for ancestor in destination
        .ancestors()
        .take_while(|p| !p.as_os_str().is_empty())
    {
        match fs::canonicalize(ancestor) {
            Ok(base) => {
                found_base = Some((base, ancestor));
                break;
            }
            Err(err) if err.kind() == ErrorKind::NotFound => {
                // keep walking up
            }
            Err(err) => return Err(err),
        }
    }

    let (base, tail) = if let Some((base, ancestor)) = found_base {
        let tail = destination.strip_prefix(ancestor).unwrap_or(Path::new(""));
        (base, tail)
    } else {
        let base = fs::canonicalize(std::env::current_dir()?)?;
        (base, destination)
    };

    Ok(apply_tail(base, tail))
}
```

Key benefits:

- Single, linear control flow for missing-destination handling.
- No reversed iteration or bespoke `OsString`-based reconstruction.
- `..` handling is explicit and localized to `apply_tail`, using standard `Component` APIs.
</issue_to_address>

### Comment 3
<location path="docs/execplans/9-7-1-first-party-policy-hint-resolution.md" line_range="32" />
<code_context>
+rstest_bdd_harness_tokio::TokioHarness
+```
+
+This matters because ADR-008 makes `harness = ...` the lead configuration for
+first-party integrations. Users should be able to select `StdHarness`,
+`TokioHarness`, or `GpuiHarness` without also repeating the matching
</code_context>
<issue_to_address>
**suggestion (review_instructions):** The acronym “ADR-008” is introduced without expanding ADR on first use.

The document refers to “ADR-008” here without first expanding ADR (for example, “Architecture Decision Record (ADR) 008”). To follow the style requirement, expand ADR the first time it appears in this ExecPlan.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.md`

**Instructions:**
Define uncommon acronyms on first use.

</details>
</issue_to_address>

### Comment 4
<location path="docs/execplans/9-7-1-first-party-policy-hint-resolution.md" line_range="360" />
<code_context>
+  `StdHarness`.
+- `crates/rstest-bdd-harness-tokio`: first-party Tokio harness and
+  `TokioAttributePolicy`.
+- `crates/rstest-bdd-harness-gpui`: first-party GPUI harness and
+  `GpuiAttributePolicy`.
+
</code_context>
<issue_to_address>
**suggestion (review_instructions):** The acronym “GPUI” is used here without being expanded on first use in this document.

This is the first time GPUI appears in this ExecPlan, but it is not expanded (for example, “GPUI (Graphical/GUI-related…whatever is correct in this project)”). Please expand GPUI on first use to satisfy the acronym rule.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.md`

**Instructions:**
Define uncommon acronyms on first use.

</details>
</issue_to_address>

### Comment 5
<location path="docs/execplans/9-7-1-first-party-policy-hint-resolution.md" line_range="228" />
<code_context>
+  `resolve_attribute_policy` even though the functions exist and compile.
+  Evidence: direct `leta show` invocations failed before source inspection.
+  Impact: continue using `rg` and targeted file reads for this small Rust
+  change, while recording the LSP limitation.
+- Observation: the existing code already implements the shared resolver
+  surface and macro precedence wiring requested by 9.7.1. Evidence:
</code_context>
<issue_to_address>
**suggestion (review_instructions):** The acronym “LSP” is introduced without definition on first use.

“LSP” appears here without being defined (for example, “language server protocol (LSP)”). Please expand the acronym at its first occurrence in the document to comply with the acronym definition requirement.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.md`

**Instructions:**
Define uncommon acronyms on first use.

</details>
</issue_to_address>
~~~

**Comment by leynos** [comment]
@coderabbitai Please investigate the cause of the following issue using codegraph exploration and research, identify a fix and provide an AI coding agent prompt for the fix:

~~~
Run leynos/shared-actions/.github/actions/generate-coverage@aebb3f5b831102e2a10ef909c83d7d50ea86c332
  with:
    output-path: lcov.info
    format: lcov
    features: rstest-bdd/diagnostics rstest-bdd-macros/compile-time-validation rstest-bdd-server/test-support
    with-default-features: true
    use-cargo-nextest: false
    with-ratchet: false
    baseline-rust-file: .coverage-baseline.rust
    baseline-python-file: .coverage-baseline.python
    with-cucumber-rs: false
  env:
    CARGO_TERM_COLOR: always
    CS_ACCESS_TOKEN: 
    CODESCENE_CLI_SHA256: 
    RUSTFLAGS: -D warnings
    RUSTDOCFLAGS: -D warnings
    CARGO_INCREMENTAL: 0
    CARGO_PROFILE_DEV_DEBUG: 0
    RUST_BACKTRACE: short
    CARGO_UNSTABLE_SPARSE_REGISTRY: true
    CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
    UV_TOOL_BIN_DIR: D:\a\_temp\uv-tool-bin-dir
    UV_TOOL_DIR: D:\a\_temp\uv-tool-dir
    UV_CACHE_DIR: D:\a\_temp\setup-uv-cache
    SCCACHE_PATH: C:\hostedtoolcache\windows\sccache\0.15.0\x64/sccache
    ACTIONS_CACHE_SERVICE_V2: on
    ACTIONS_RESULTS_URL: https://results-receiver.actions.githubusercontent.com/
    ACTIONS_RUNTIME_TOKEN: ***
    UV_PYTHON_INSTALL_DIR: D:\a\_temp\uv-python-dir
...
test std_harness::tests::std_harness_propagates_runner_panics ... ok
test trybuild_staging::tests::copy_dir_tree_rejects_missing_tail_overlap_destinations::case_2_resolved_back_to_source ... FAILED
test error::tests::runtime_build_failed_display_snapshot ... ok

failures:

---- trybuild_staging::tests::copy_dir_tree_rejects_missing_tail_overlap_destinations::case_2_resolved_back_to_source stdout ----

thread 'trybuild_staging::tests::copy_dir_tree_rejects_missing_tail_overlap_destinations::case_2_resolved_back_to_source' (2936) panicked at crates\rstest-bdd-harness\src\trybuild_staging\tests.rs:217:9:
dst should not be created
stack backtrace:
   0: std::panicking::panic_handler
             at /rustc/59807616e1fa2540724bfbac14d7976d7e4a3860/library\std\src\panicking.rs:689
   1: core::panicking::panic_fmt
             at /rustc/59807616e1fa2540724bfbac14d7976d7e4a3860/library\core\src\panicking.rs:80
   2: rstest_bdd_harness::trybuild_staging::tests::copy_dir_tree_rejects_missing_tail_overlap_destinations
   3: <alloc::boxed::Box<dyn core::ops::function::FnOnce<((),), Output = !>> as core::ops::function::FnOnce<((),)>>::call_once
   4: core::ops::function::FnOnce::call_once
             at /rustc/59807616e1fa2540724bfbac14d7976d7e4a3860/library\core\src\ops\function.rs:250
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    trybuild_staging::tests::copy_dir_tree_rejects_missing_tail_overlap_destinations::case_2_resolved_back_to_source

test result: FAILED. 27 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.05s

error: test failed, to rerun pass `-p rstest-bdd-harness --lib`
error: process didn't exit successfully: `'C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin\cargo.exe' test --tests --manifest-path 'D:\a\rstest-bdd\rstest-bdd\Cargo.toml' --target-dir 'D:\a\rstest-bdd\rstest-bdd\target\llvm-cov-target' --workspace --features 'rstest-bdd/diagnostics rstest-bdd-macros/compile-time-validation rstest-bdd-server/test-support'` (exit code: 101)
cargo llvm-cov --workspace --summary-only --features 'rstest-bdd/diagnostics rstest-bdd-macros/compile-time-validation rstest-bdd-server/test-support' --lcov --output-path lcov.info failed with code 101
Error: Process completed with exit code 101.
~~~

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### crates/rstest-bdd-harness/src/trybuild_staging/prop_tests.rs (1)

`1-1`: _🧹 Nitpick_ | _🔵 Trivial_ | _⚡ Quick win_
**Update the module synopsis to match the expanded property-test scope.**
Rewrite the `//!` comment so it documents symlink rejection _and_ missing-destination canonicalization/overlap invariants now covered in this module.

##### ✏️ Proposed patch

```diff
-//! Property tests for symlink rejection in [`super::copy_dir_tree`].
+//! Property tests for `copy_dir_tree` safety invariants.
+//!
+//! This module covers symlink rejection and destination canonicalization /
+//! overlap handling for missing-parent and `..` path patterns.
```

As per coding guidelines, “Every module **must** begin with a module level (`//!`) comment explaining the module's purpose and utility”.

##### 🤖 Prompt for AI Agents
